modules/social_features/social_profile/modules/social_profile_preview/social_profile_preview.module

```diff
@@ -136,28 +136,38 @@ function social_profile_preview_preprocess_flag(array &$variables): void {
  * Implements hook_preprocess_HOOK().
  */
 function social_profile_preview_preprocess_profile(array &$variables): void {
-  $view_mode = $variables['elements']['#view_mode'];
-
-  if (in_array($view_mode, ['compact_teaser', 'small_teaser'])) {
-    /** @var \Drupal\profile\Entity\ProfileInterface $profile */
-    $profile = $variables['profile'];
-
-    /** @var \Drupal\social_profile_preview\Service\SocialProfilePreviewHelperInterface $helper */
-    $helper = \Drupal::service('social_profile_preview.helper');
-
-    $helper->alter(
-      $profile,
-      $variables,
-      'link_attributes',
-      TRUE,
-      'profile_name',
-      'profile_name_extra',
-    );
-
-    if ($view_mode === 'compact_teaser') {
-      $helper->alter($profile, $variables, 'avatar_attributes', TRUE);
-    }
-  }
+//  $view_mode = $variables['elements']['#view_mode'];
+
+  $profile = $variables['elements']['#profile'];
+
+  if (in_array($variables['view_mode'], ['compact', 'compact_notification'])) {
+    $preview_url = Url::fromRoute('social_profile_preview.canonical', [
+      'profile' => $profile->get('profile_id')->value,
+    ])->toString();
```

Comment:
to have a URL without prefixes, you can use the `getInternalPath() `method instead of the `toString()`